pkg/3scale/amp/operator/deployment_config_reconciler.go

```diff
@@ -4,228 +4,108 @@ import (
-	// our logic
-	// TODO
-
-	// reconcile update
-	needsUpdate, err := r.ensureDeploymentConfig(existingDeploymentConfig, desiredDeploymentConfig)
-	if err != nil {
-		return err
-	}
-
-	if needsUpdate {
-		r.Logger().Info(fmt.Sprintf("Updating DeploymentConfig %s", objectInfo))
-		err := r.Client().Update(context.TODO(), existingDeploymentConfig)
-		if err != nil {
-			r.Logger().Error(err, fmt.Sprintf("error updating DeploymentConfig %s", objectInfo))
-			return err
-		}
+	if r.reconciler.IsUpdateNeeded(desiredDC, existingDC) {
+		return r.updateResource(existingDC)
 	}
 
 	return nil
 }
 
-func (r *DeploymentConfigReconciler) ensureDeploymentConfig(updated, desired *appsv1.DeploymentConfig) (bool, error) {
-	changed := false
+func DeploymentConfigReconcileContainerResources(desired, existing *appsv1.DeploymentConfig, logger logr.Logger) bool {
+	desiredName := ObjectInfo(desired)
+	update := false
 
-	objectMetaChanged, err := r.ObjectMetaMerger.EnsureObjectMeta(&updated.ObjectMeta, &desired.ObjectMeta)
-	if err != nil {
-		return false, err
-	}
-	if objectMetaChanged {
-		changed = true
-	}
-
-	// Reconcile PodTemplateSpec labels
-	if r.ObjectMetaMerger.EnsureLabels(updated.Spec.Template.ObjectMeta.Labels, desired.Spec.Template.ObjectMeta.Labels) {
-		changed = true
+	//
+	// Check container resource requirements
+	//
+	if len(desired.Spec.Template.Spec.Containers) != 1 {
+		panic(fmt.Sprintf("%s desired spec.template.spec.containers length changed to '%d', should be 1", desiredName, len(desired.Spec.Template.Spec.Containers)))
 	}
 
-	r.ensurePodContainers(updated.Spec.Template.Spec.Containers, desired.Spec.Template.Spec.Containers)
-	if !reflect.DeepEqual(updated.Spec.Template.Spec.Containers, desired.Spec.Template.Spec.Containers) {
-		fmt.Println(cmp.Diff(updated.Spec.Template.Spec.Containers, desired.Spec.Template.Spec.Containers, cmpopts.IgnoreUnexported(resource.Quantity{})))
-		updated.Spec.Template.Spec.Containers = desired.Spec.Template.Spec.Containers
-		changed = true
+	if len(existing.Spec.Template.Spec.Containers) != 1 {
+		logger.Info(fmt.Sprintf("%s spec.template.spec.containers length changed to '%d', recreating dc", desiredName, len(existing.Spec.Template.Spec.Containers)))
+		existing.Spec.Template.Spec.Containers = desired.Spec.Template.Spec.Containers
+		update = true
 	}
 
-	r.ensurePodContainers(updated.Spec.Template.Spec.InitContainers, desired.Spec.Template.Spec.InitContainers)
-	if !reflect.DeepEqual(updated.Spec.Template.Spec.InitContainers, desired.Spec.Template.Spec.InitContainers) {
-		fmt.Println(cmp.Diff(updated.Spec.Template.Spec.InitContainers, desired.Spec.Template.Spec.InitContainers, cmpopts.IgnoreUnexported(resource.Quantity{})))
-		updated.Spec.Template.Spec.InitContainers = desired.Spec.Template.Spec.InitContainers
-		changed = true
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.Containers[0].Resources, desired.Spec.Template.Spec.Containers[0].Resources) {
```

Comment:
This comparison causes false positives due to reflect ends comparing unexported resources and the way resourceRequirements are internally stored sometimes differs the way that we set it.
To solve that problem in one of my original commits I implemented a solution for that that you can reuse to fix this false positives. The method where I implemented was called "ensureResourceRequirements"

Reply:
:+1: